input-explainer.md

```diff
@@ -280,6 +292,91 @@ When the canvas receives a `pointerdown` event an `XRInputSource` is created wit
 
 For each of these events the `XRInputSource`'s target ray must be updated to originate at the point that was interacted with on the canvas, projected onto the near clipping plane (defined by the `depthNear` attribute of the `XRSession`) and extending out into the scene along that projected vector.
 
+## Button and Axis State
+
+Some applications need more than point-and-click style interaction provided by the `select` events. For devices that make use of controllers with buttons and axes, more complete information about the state of those inputs can be observed via the `XRInputSource`'s `gamepad` attribute.
+
+If the `XRInputState` represents an XR controller device with buttons or axes the `gamepad` attribute is an instance of a [`Gamepad`](https://w3c.github.io/gamepad/#gamepad-interface) interface, and `null` otherwise. Examples of controllers that may expose their state this way include Oculus Touch, Vive wands, Oculus Go and Daydream controllers, or other similar devices. Controllers not directly associated with the XR device, such as the majority of traditional gamepads, should not be exposed using this interface. Tracked devices without discreet inputs, such as optical hand tracking, should also have a `null` `gamepad` attribute.
+
+`Gamepad` instances reported in this way have several notable behavioral changes vs. the ones reported by `navigator.getGamepads()`:
+
+  - `Gamepad` instances connected to an `XRInputSource` must not be included in the array returned by `navigator.getGamepads()`.
+  - The `Gamepad`'s `index` attribute must be `-1`.
+  - The `Gamepad`'s `connected` attribute must always be `true`.
+  - The `Gamepad`'s `mapping` attribute must be `xr-standard`.
+
+Finally, the `id` attribute for `Gamepad`s surfaced by the WebXR API are more strictly formatted than those of traditional gamepads in order to make them a more appropriate key for determining rendering assets.
+
+  - The `id` MAY be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason. Applications should render a generic input device in this case.
+  - Inline sessions MUST only expose `id`s of `'unknown'`.
```

Comment:
This is an interesting question... are we going to allow `tracked-pointer` input sources for `inline` sessions?

Reply:
Probably not, but as indicated above I don't know that we should restrict this to only `tracked-pointer` input sources. Admittedly, though, that's mostly because there's a couple of examples of `gaze` devices that have buttons/axes. I have a hard time seeing how this would be applicable to `screen` input sources.